### PR TITLE
SSCS-9228 Upload document FE hearing recording request

### PR DESCRIPTION
**Pull request by jackmaloney**
https://tools.hmcts.net/jira/browse/SSCS-9228

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/uploaddocuments/UploadDocumentFurtherEvidenceMidEventHandler.java
Comment (ryandac) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/uploaddocuments/UploadDocumentFurtherEvidenceMidEventHandler.java:54]: nice to have: use the functional style to get the requestHearingCount i.e. stream, filter, sum.
nice to have: put this in a separate method that describes what it does.

File: src/main/java/uk/gov/hmcts/reform/sscs/service/admin/RestoreCasesService2.java
Comment (ryandac) [inline src/main/java/uk/gov/hmcts/reform/sscs/service/admin/RestoreCasesService2.java]: not a great api is it? #whatever!

File: src/main/resources/config/application.properties
Comment (ryandac) [inline src/main/resources/config/application.properties:38]: Ctrl-Z please

File: src/main/java/uk/gov/hmcts/reform/sscs/util/SscsUtil.java
Comment (ryandac) [inline src/main/java/uk/gov/hmcts/reform/sscs/util/SscsUtil.java:19]: Nice to have: This is quite a confusing api. Maybe have multiple different methods so we can remove the boolean parameters.

**Review by ryandac — CHANGES_REQUESTED** [review]
Remove the changes to the application.properties.
Added nice to haves suggestions, not required for approval.

File: src/main/java/uk/gov/hmcts/reform/sscs/service/admin/RestoreCasesService2.java
Comment (jackmaloney) [reply src/main/java/uk/gov/hmcts/reform/sscs/service/admin/RestoreCasesService2.java]: This wasn't my change - there was a breaking commit to common so just added this until another PR was merged. Looks like it's in now so this should disappear from my PR

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/uploaddocuments/UploadDocumentFurtherEvidenceMidEventHandler.java
Comment (jackmaloney) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/uploaddocuments/UploadDocumentFurtherEvidenceMidEventHandler.java:54]: Good shout - updated

File: src/main/java/uk/gov/hmcts/reform/sscs/util/SscsUtil.java
Comment (jackmaloney) [reply src/main/java/uk/gov/hmcts/reform/sscs/util/SscsUtil.java:19]: Split this into 2 methods now

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/uploaddocuments/UploadDocumentFurtherEvidenceMidEventHandler.java
Comment (ryandac) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/uploaddocuments/UploadDocumentFurtherEvidenceMidEventHandler.java:64]: Nice to have:
Functional programming removes the need to have mutable variables, i.e. requestHearingCount, could use emptyIfNull instead of null check.
put filter and count on new lines so that it is readable

**Review by ryandac — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/uploaddocuments/UploadDocumentFurtherEvidenceMidEventHandler.java
Comment (jackmaloney) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/uploaddocuments/UploadDocumentFurtherEvidenceMidEventHandler.java]: Done

File: src/main/java/uk/gov/hmcts/reform/sscs/service/admin/RestoreCasesService2.java
Comment (ryandac) [reply src/main/java/uk/gov/hmcts/reform/sscs/service/admin/RestoreCasesService2.java]: great :)

**Pull request merged**
